nexus/db-queries/src/db/datastore/region_snapshot_replacement.rs

```diff
@@ -1288,18 +1330,31 @@ mod test {
             RegionSnapshotReplacementStep::new(Uuid::new_v4(), volume_id);
         let first_request_id = step.id;
 
-        datastore
+        let result = datastore
             .insert_region_snapshot_replacement_step(&opctx, step)
             .await
             .unwrap();
 
+        assert!(matches!(
+            result,
+            InsertRegionSnapshotReplacementStepResult::Inserted { .. }
+        ));
+
         let step =
             RegionSnapshotReplacementStep::new(Uuid::new_v4(), volume_id);
 
-        datastore
+        let result = datastore
             .insert_region_snapshot_replacement_step(&opctx, step.clone())
             .await
-            .unwrap_err();
+            .unwrap();
+
+        let InsertRegionSnapshotReplacementStepResult::AlreadyHandled {
+            existing_step_id,
+        } = result
+        else {
+            panic!("wrong return type");
```

Comment:
could we have this panic include `result` so that if we hit it, the test's failure output includes the unexpected value?

Reply:
good idea, done in d2135f21b